### ISSUE TITLE
Request for list pattern providing keyboard navigation

### ISSUE THREAD

**Issue by pereng11** [w3c/aria-practices#2681]
## Current situation

Hello. I have a question for implementing content list with appropriate keyboard navigation.

I want to make List component that provide appropriate keyboard navigation. Because contents in nested list takes too long time to navigate. I referred APG, but there's no exactly matched pattern for my case.

So I read [Developing a Keyboard Interface](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/) and understood managing tab sequence hierarchically is basic principle for keyboard navigation. 

![Large GIF (1154x622)](https://user-images.githubusercontent.com/67821588/234515045-0746b66c-5ae0-48af-9bd0-90deb4292307.gif)

This is example of navigating contents in nested list I made. In this example, tab sequence is `days - webtoons in day - outside of component`. Each content in the list is link to webtoon detail page.

These are description of each key.

- `Tab`: Move to next hierarchy. If activeElement is the last focusable element in component, escape focus to outside of component.
- `Shift+Tab`: Move to previous hierarchy. If activeElement is the first focusable element in component, escape focus to outside of component.
- `↑/←`: Move focus to previous focusable element in the level. If activeElement is the first element in the level, focus doesn't move.
- `↓/→`: Move focus to next focusable element in the level. If activeElement is the last element in the level, focus doesn't move.
- `Home(In macOS, ⌘+↑/←)`: Move focus to the first focusable element in the level.
- `End(In macOS, ⌘+↓/→)`: Move focus to the last focusable element in the level.
- `Ctrl+Home`: Move focus to the first focusable element in the component.
- `Ctrl+End`: Move focus to the last focusable element in the component.
- `PageUp(In macOS, ⌥+↑/←)`: Move focus to forward by step in the level.
- `PageDown(In macOS, ⌥+↓/→)`: Move focus to backward by step in the level.

`Arrow key` is used for moving in same level, between days of the week or between webtoons in the day.
`Tab` and `Shift+Tab` is used for moving hierarchy, days <-> webtoons <-> outside of component.
Special keys like `Home`, `End` is used to move more quickly.

## Problem

I mainly referred [layout gird example](https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/layout-grids/) but there's some difference between content list and layout grid.

In grid, all items are in the same group. So all arrow keys are used matching the visual orientation.
But In content list, contents belong to each group. So only keys that match the direction of the list are used(only left&right or up&down). Furthermore, I thought arrow keys should not used to move focus between contents in different groups in perspective of document structure. For example, move from webtoon in friday to webtoon in saturday using arrow key shouldn't be accepted.

So I made keyboard navigation different from the grid guide, but I'm not sure if this is right.

## In conclusion

I wonder

**1. Why there's no example for content list pattern**
**2. Is there any plan or discussion about these cases**
**3. keyboard navigation that I made is appropriate**

**Comment by mcking65** [comment]
Hi @pereng11 

I am not sure if I fully understand your example because I can't see the image. I am also not sure I understand why you need to provide keyboard navigation of the list. Are the webtoons interactive? If they are, how many interactions are there for each webtoon? How are they performed? Is each webtoon like a single button?

There have been discussions of hierarchical interactive lists. So far, the only complete solution is provided by [treegrid](https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/).listhis type of

**Comment by pereng11** [comment]
Thank you for comment @mcking65.

There's no image problem in my computer, so I added [deployed storybook site link](https://improve-a11y.du.r.appspot.com/?path=/docs/naver-webtoon-alldaylist--docs) below the image. You can check example and keyboard navigation yourself there. Each webtoon in webtoon list has link to webtoon's episode list page.

I read `treegrid` pattern but I'm not sure about keyboard interaction.

In my case(nested list), `Tab/Shift+Tab` key is used to move inside or outside of list because I thought it is appropriate in tab sequence point of view. But in `treegrid` pattern, `Arrow keys` are used to move inside or outside of list and `Tab/Shift+Tab` key is used to Enter/escape the whole component. 
It seems reasonable, but I think treegrid's method can't cover all cases. When direction of outer and inner list are same, arrow keys role is ambiguous because it's hard to decide correct move between move in outer list and move in inner list.

So I think nested list is different case from current grid or treegrid, what do you think about it?
